src/NServiceBus.Core/Timeout/Hosting/Windows/TimeoutDispatcherProcessor.cs

```diff
@@ -32,7 +32,16 @@ public bool Handle(TransportMessage message)
 
             if (TimeoutsPersister.TryRemove(timeoutId, out timeoutData))
             {
-                MessageSender.Send(timeoutData.ToTransportMessage(), timeoutData.ToSendOptions(Configure.LocalAddress));
+                try
+                {
+                    MessageSender.Send(timeoutData.ToTransportMessage(), timeoutData.ToSendOptions(Configure.LocalAddress));
+                }
+                catch
+                {
+                    timeoutData.Id = null;
+                    timeoutData.Time = DateTime.UtcNow.AddSeconds(5);
+                    TimeoutsPersister.Add(timeoutData);
```

Comment:
@andreasohlund indeed we're just reducing the chance of the message loss happening. But not sure there is a real alternative besides breaking interfaces or touching every timeout persistence (which also breaks the interface...).